File: Logitude/AngularModules/AngularModules/Infrastructure/EntityPMs/ScreenPM.ts

export class ScreenPM {

    private id: string;
    public get Id() { return this.id; }
    public set Id(newValue: string) { this.id = newValue;}


    private tenant: number;
    public get Tenant() { return this.tenant; }
    public set Tenant(newValue: number) { this.tenant = newValue; }

    private numberOfRows: number;
    public get NumberOfRows() { return this.numberOfRows; }
    public set NumberOfRows(newValue: number) { this.numberOfRows = newValue; }

    private numberOfColumns: number;
    public get NumberOfColumns() { return this.numberOfColumns; }
    public set NumberOfColumns(newValue: number) { this.numberOfColumns = newValue; }

    private userTenant: number;
    public get UserTenant() { return this.userTenant; }
    public set UserTenant(newValue: number) { this.userTenant = newValue; }

    private name: string;
    public get Name() { return this.name; }
    public set Name(newValue: string) { this.name = newValue; }

    private code: string;
    public get Code() { return this.code; }
    public set Code(newValue: string) { this.code = newValue; }


    private objectTableId: string;
    public get ObjectTableId() { return this.objectTableId; }
    public set ObjectTableId(newValue: string) { this.objectTableId = newValue; }


    private objectTableName: string;
    public get ObjectTableName() { return this.objectTableName; }
    public set ObjectTableName(newValue: string) { this.objectTableName = newValue; }


    private isReadOnly: boolean;
    public get IsReadOnly() { return this.isReadOnly; }
    public set IsReadOnly(newValue: boolean) { this.isReadOnly = newValue; }

    private inactive: boolean;
    public get Inactive() { return this.inactive; }
    public set Inactive(newValue: boolean) { this.inactive = newValue; }


    private type: string;
    public get Type() { return this.type; }
    public set Type(newValue: string) { this.type = newValue; }


    private sortedByFieldCode: string;
    public get SortedByFieldCode() { return this.sortedByFieldCode; }
    public set SortedByFieldCode(newValue: string) { this.sortedByFieldCode = newValue; }


    private sortedType: string;
    public get SortedType() { return this.sortedType; }
    public set SortedType(newValue: string) { this.sortedType = newValue; }

    private relatedScreenCode: string;
    public get RelatedScreenCode() { return this.relatedScreenCode; }
    public set RelatedScreenCode(newValue: string) { this.relatedScreenCode = newValue; }

    public OldEntityPM: ScreenPM;

    private isDirty: boolean;
    public get IsDirty() { return this.isDirty; }
    public set IsDirty(newValue: boolean) { this.isDirty = newValue; }

    private disableMarkAsDirty: boolean;
    public get DisableMarkAsDirty() { return this.disableMarkAsDirty; }
    public set DisableMarkAsDirty(newValue: boolean) { this.disableMarkAsDirty = newValue; }

    private isHeaderScreen: boolean;
    public get IsHeaderScreen() { return this.isHeaderScreen; }
    public set IsHeaderScreen(newValue: boolean) { this.isHeaderScreen = newValue; }

}

File: AmitalCloud/AmitalCloud.Infrastructure.Domain/EntityPMs/Manual/ScreenPM.cs
using AmitalCloud.Infrastructure.Domain.BaseClasses;
using AmitalCloud.Infrastructure.Domain.EntityPOCOs;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ServiceModel.DomainServices.Server;

namespace AmitalCloud.Infrastructure.Domain.EntityPMs
{
    public class ScreenPM : BaseEntityPM
    {
        public ScreenPM() : base() { }
        public ScreenPM(Screen entity) : base()
        {
            Id = entity.Id;
            Code = entity.Code;
            NumberOfRows = entity.NumberOfRows;
            NumberOfColumns = entity.NumberOfColumns;
            ObjectTableId = entity.ObjectTableId;
            Tenant = entity.Tenant;
            IsReadOnly = entity.IsReadOnly;
            Inactive = entity.Inactive;
            ObjectTableName = entity.ObjectTable?.Name;
            Name = entity.Name;
            //UserTenant = entity..UserTenant;
            Type = entity.Type;
            SortedByFieldCode = entity.SortedByFieldCode;
            SortedType = entity.SortedType;
            SearchFields = entity.SearchFields;
            //ChildScreenGrid = entity..ChildScreenGrid;
            RelatedScreenCode = entity.RelatedScreenCode;
            IsHeaderScreen = entity.IsHeaderScreen;
            ScreenFields = new List<ScreenFieldPM>();
            //foreach (var item in entity..ScreenFields)
            //{
            //    ScreenFields.Add(new ScreenFieldPM(item));
            //}
        }   
        private Screen headerScreen;


        [Key]
        public string Id { get; set; }


        public string Code { get; set; }

        public int NumberOfRows { get; set; }
        public int NumberOfColumns { get; set; }

        public string ObjectTableId { get; set; }

        public int Tenant { get; set; }

        public bool IsReadOnly { get; set; }
        public bool Inactive { get; set; }

        public string ObjectTableName { get; set; }
        public string Name { get; set; }
        public int UserTenant { get; set; }
        public string Type { get; set; }
        public string SortedByFieldCode { get; set; }
        public string SortedType { get; set; }
        public string SearchFields { get; set; }
        public string ChildScreenGrid { get; set; }
        public string RelatedScreenCode { get; set; }
        public bool IsHeaderScreen { get; set; }
        [Include]
        [Association("ScreenScreenField", "Id", "ScreenId")]
        public virtual List<ScreenFieldPM> ScreenFields { get; set; }

    }
}